### ISSUE TITLE
Hole horizontal expansion causes hole deformation

### ISSUE THREAD

**Issue by cicinovec** [Ultimaker/Cura#11369]
### Application Version

4.13.0

### Platform

Windows 10

### Printer

Custom

### Reproduction steps

1. Slice simple model with round hole with Hole horizontal expansion 0 mm or positive => result is round hole
2. Slice the same model with Hole horizontal expansion -0.1 mm (negative) => result is star shaped hole with deformed shape

### Actual results

1. Nice round hole
2. Deformed hole

### Expected results

Round hole with different diameter only

### Checklist of files to include

- [X] Log file
- [X] Project file

### Additional information & file uploads

![Hole_expansion_negative](https://user-images.githubusercontent.com/33634908/151650580-429443c7-db99-4a22-a218-4974576891cd.png)
![Hole_expansion_0](https://user-images.githubusercontent.com/33634908/151650582-17fac280-6300-483b-8d2f-0946308c2153.png)
[Helping_hand_clamp_PLA_T1.zip](https://github.com/Ultimaker/Cura/files/7963124/Helping_hand_clamp_PLA_T1.zip)
[cura.log](https://github.com/Ultimaker/Cura/files/7963125/cura.log)

**Comment by GregValiant** [comment]
When I opened the STL and sliced it the results were about the same as you show in your images.
I opened the 3mf file and then exported the model as an STL so I could check it for errors.  None were found.
I cleared the build plate and brought in the STL (Cura 4.13.1).  There were no problems with the HHE at -0.05.  With the HHE at -0.2 there was a very slight inconsistency of the Inner-Walls around the hole.  At -0.4 it went away.
This is using your settings and at HHE = -0.05.
![image](https://user-images.githubusercontent.com/64202104/151662496-e58ba490-3882-4667-b621-3087d59675de.png)

This is with HHE at -0.2
![image](https://user-images.githubusercontent.com/64202104/151662349-5dc36f8f-64c3-40b9-aedf-7f7641edb876.png)

This image is at HHE -0.4.
![image](https://user-images.githubusercontent.com/64202104/151662308-537d22d1-857c-451e-b3c2-5ae1a26fa988.png)
